packages/adminrouter/extra/src/README.md

```diff
@@ -261,31 +261,188 @@ as-is/usually do not follow any of these patterns. It is possible to standardise
-
-In order for the forwarding to work reliably across task failures, we recommend co-locating the endpoints with the task. This way, if the task is restarted on a potentially other host and with different ports, Admin Router will pick up the new labels and update the routing. NOTE: Due to caching there might be an up to 30-second delay until the new routing is working.
-
-We would recommend having only a single task setting these labels for a given `service-name`.
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+The `/service` endpoint supports WebSockets, by default disables NGINX request
+and response buffering, all redirects from services are rewritten in order to
+support the `/service` URL path prefix clients are using.
+
+### Intended use
+It is important to remember that the `/service` endpoint was designed to work
+with root Marathon only. Tasks launched by Marathon-On-Marathon instances will
+not be reachable via it.
+
+In the case of Mesos frameworks, this becomes a bit more complex - all the
+frameworks are visible in Mesos' `/state-summary` endpoint output so in theory
+it is possible to access a framework launched using MoM or by hand.
+Unfortunately, Mesos as of now does not enforce framework name uniqueness and
+thus it is possible to launch multiple frameworks with the same name. AR in
+such case will route in a non-deterministic manner. Root Marathon by default
+enforces unique names for all the tasks and frameworks it has under control.
+
+Please check the [Limitations](#limitations) section for more details.
+
+### Operation
+Under the hood, the /service endpoint implementation uses non-trivial logic in
+order to resolve the cluster-internal socket address to route the HTTP request
+to.
+
+#### Recursive resolving
+There is no way to tell which part of the path is application ID and
+which one is just a resource address. For example, assuming that the request
+path is `/foo/bar/baz/picture.jpg` we can have:
+* namespace and name: `foo`, resource address: `/bar/baz/picture.jpg`
+* namespace and name: `foo/bar`, resource address: `/baz/picture.jpg`
+* namespace and name: `foo/bar/baz`, resource address: `/picture.jpg`
+* namespace and name: `foo/bar/baz/picture.jpg`, resource address: `/`
+
+So the algorithm iteratively tries to resolve each "candidate" namespace and
+name and assumes that first-match wins.
+
+Assuming that we deployed `foo/bar` NGINX task, we will see two iterations:
```

Comment:
❓ "an NGINX task named `foo/bar`

or at least some rewording as this doesn't flow